src/sentry/migrations/0890_split_discover_dataset_saved_queries.py

```diff
@@ -0,0 +1,111 @@
+# Generated by Django 5.2.1 on 2025-05-14 13:57
+
+
+from enum import Enum
+
+from django.db import migrations
+from django.db.backends.base.schema import BaseDatabaseSchemaEditor
+from django.db.migrations.state import StateApps
+
+from sentry.discover.dataset_split import _get_and_save_split_decision_for_query
```

Comment:
Be careful with importing application code into migrations. You'll need to keep this function around (and keep the same interface) for as long as this migration lives, which could be several years.

In other migrations we duplicate the imported functions into the migration so that we don't accidentally break compatibility.

Reply:
The imported function has a lot of dependencies on the existing application code, so it's not going to be easy to duplicate it into the migration. The only way around it is to have a hard stop upgrade https://develop.sentry.dev/self-hosted/releases/#hard-stops and no-op this migration when we want to remove the code.

Reply:
Ok. Having a test for the migration will help catch any breaks. A hard stop is an option as well.